### ISSUE TITLE
Microsoft Outlook connection Test Message

### ISSUE THREAD

**Issue by JM-Lemmi** [cirosec-studis/zimbraweb-smtp-bridge#24]
The Milter needs to allow the Microsoft Outlook connection Test Message, which is sadly multipart, otherwise setup will always fail on Outlook:

```
Nov 11 21:25:28 c6c450bc8682 postfix/postfix-script[81]: starting the Postfix mail system
Starting the Zimbra Milter
Changing chmod of socket to 777
2021Nov11 21:27:38 [1]
connect from [172.17.0.1] at ('172.17.0.1', 57806)

2021Nov11 21:27:38 [1]
hello from JulianX380

2021Nov11 21:27:41 [1]
user
s212689
sent mail from
<s212689@student.dhbw-mannheim.de>
()

2021Nov11 21:27:41 [1]
rcpt to
<s212689@student.dhbw-mannheim.de>
()

2021Nov11 21:27:41 [1]
Subject: b'Microsoft Outlook-Testnachricht'

From: "Microsoft Outlook" <s212689@student.dhbw-mannheim.de>
Sender: "Microsoft Outlook" <s212689@student.dhbw-mannheim.de>
To: "Julian Lemmerich" <s212689@student.dhbw-mannheim.de>
Subject: Microsoft Outlook-Testnachricht
MIME-Version: 1.0
Content-Type: multipart/alternative;
boundary="----=_NextPart_000_0000_01D7D74B.52629370"

This is a multipart message in MIME format.

------=_NextPart_000_0000_01D7D74B.52629370
Content-Type: text/plain;
charset="utf-8"
Content-Transfer-Encoding: quoted-printable

Diese E-Mail-Nachricht wurde von Microsoft Outlook automatisch =
w=C3=A4hrend des Testens der Kontoeinstellungen gesendet.

------=_NextPart_000_0000_01D7D74B.52629370
Content-Type: text/html;
charset="utf-8"
Content-Transfer-Encoding: quoted-printable

<html><body><p>Diese E-Mail-Nachricht wurde von Microsoft Outlook =
automatisch w=C3=A4hrend des Testens der Kontoeinstellungen gesendet.
</p></body></html>

------=_NextPart_000_0000_01D7D74B.52629370--
```

![image](https://user-images.githubusercontent.com/41118534/141371840-084bc59f-ee99-492a-8ec3-ac3a6245f1cf.png)

# ~~current workaround~~

~~use the alpha.1 container to setup then switch to the newer container~~

**Comment by frereit** [comment]
Oh but look at that, Outlook correctly displays the SMTP error. So this closes #23 as it confirms an issue with thunderbird. I've been stuck on "Uninstalling Outlook" for the last half hour so I couldn't verify. 

We can add multipart/alternative support. The reason why I didn't want to support it is because sometimes the `text/plain` part can just be "Please look at the HTML", which is at no use if there's no HTML. We *could* convert just strip out the HTML and attach the original mail as an eml? Maybe that would be a nice workaround?

**Comment by JM-Lemmi** [comment]
I feel like that would get us caught in a lot of spam filters. Writing a plain message with eml attached is very spam like.

I have the same worry. The cheekiest workaround would be to only accept the Outlook-Testmessage in multipart/alternative. This keeps current behaviour for normal mails but allows the setup to work. In my opinion the least risk while allowing Outlook Setup.

**Comment by frereit** [comment]
Good point. Yeah, let's just allow "From: "Microsoft Outlook" <s212689@student.dhbw-mannheim.de>"? I think if we only support that exact message we will run into localization issues, my windows installation is in english.

**Comment by JM-Lemmi** [comment]
Sounds like a good idea.

**Comment by frereit** [comment]
Should we add this exception-case in the zimbraweb package or in the Docker container? I feel like it'd be better in the Docker Container, we'd just have to write "our own parsing" for the multipart/mixed eml in the Docker container and then use send_raw_payload

**Comment by JM-Lemmi** [comment]
I think it belongs to the Docker container and Milter.

**Comment by frereit** [comment]
Fixed by 2017396 but please confirm and then close

**Comment by JM-Lemmi** [comment]
Confirmed working
